packages/crypto/src/crypto.ts

```diff
@@ -466,3 +568,251 @@ const bigIntToHex = (num: bigint, length: number): string => {
+ * Encrypts a private key bundle using HPKE and verifies the enclave signature.
+ *
+ * @param {string} privateKey - The private key in hexadecimal format to be encrypted.
+ * @param {string} keyFormat - The format of the private key (e.g., "SOLANA", "HEXADECIMAL").
+ * @param {any} importBundle - The bundle containing the enclave's public key and data signature for verification.
+ * @param {string} userId - The user ID that must match the signed data in the bundle.
+ * @param {string} organizationId - The organization ID that must match the signed data in the bundle.
+ * @returns {Promise<string>} - A promise that resolves to a JSON string representing the encrypted bundle.
+ * @throws {Error} - If enclave signature verification or any other validation fails.
+ */
+
+export const encryptPrivateKeyBundle = async (
+  privateKey: string,
+  keyFormat: string,
+  importBundle: any,
+  userId: string,
+  organizationId: string
+): Promise<string> => {
+  const plainTextBuf = decodeKey(privateKey, keyFormat);
+  const verified = await verifyEnclaveSignature(
+    importBundle.enclaveQuorumPublic,
+    importBundle.dataSignature,
+    importBundle.data
+  );
+  if (!verified) {
+    throw new Error(`failed to verify enclave signature: ${importBundle}`);
+  }
+
+  const signedData = JSON.parse(
+    new TextDecoder().decode(uint8ArrayFromHexString(importBundle.data))
+  );
+
+  if (
+    !signedData.organizationId ||
+    signedData.organizationId !== organizationId
+  ) {
+    throw new Error(
+      `organization id does not match expected value. Expected: ${organizationId}. Found: ${signedData.organizationId}.`
+    );
+  }
+  if (!signedData.userId || signedData.userId !== userId) {
+    throw new Error(
+      `user id does not match expected value. Expected: ${userId}. Found: ${signedData.userId}.`
+    );
+  }
+
+  if (!signedData.targetPublic) {
+    throw new Error('missing "targetPublic" in bundle signed data');
+  }
+
+  // Load target public key generated from enclave and set in local storage
+  const targetKeyBuf = uint8ArrayFromHexString(signedData.targetPublic);
+  const privateKeyBundle = hpkeEncrypt({ plainTextBuf, targetKeyBuf });
+  return formatHpkeBuf(privateKeyBundle);
+};
+
+/**
+ * Encrypts a mnemonic wallet bundle using HPKE and verifies the enclave signature.
+ *
+ * @param {string} mnemonic - The wallet mnemonic to be encrypted.
+ * @param {any} importBundle - The bundle containing the enclave's public key and data signature for verification.
+ * @param {string} userId - The user ID that must match the signed data in the bundle.
+ * @param {string} organizationId - The organization ID that must match the signed data in the bundle.
+ * @returns {Promise<string>} - A promise that resolves to a JSON string representing the encrypted wallet bundle.
+ * @throws {Error} - If enclave signature verification or any other validation fails.
+ */
+
+export const encryptWalletBundle = async (
+  mnemonic: string,
+  importBundle: any,
```

Comment:
same here. Let's have a string and document that it should be a JSON encoded string received from the activity result.

Users should just pass `initResult.bundle` straight to `encryptWalletWithBundle`; max foolproofness.